src/ILCompiler.MetadataTransform/src/ILCompiler/Metadata/Transform.Scope.cs

```diff
@@ -29,10 +29,6 @@ private ScopeDefinition HandleScopeDefinition(Cts.ModuleDesc module)
 
         private void InitializeScopeDefinition(Cts.ModuleDesc module, ScopeDefinition scopeDefinition)
         {
-            // Make sure we're expected to create a scope definition here. If the assert fires, the metadata
-            // policy should have directed us to create a scope reference (or the list of inputs was incomplete).
-            Debug.Assert(_modulesToTransform.Contains(module), "Incomplete list of input modules with respect to metadata policy");
```

Comment:
It's a bit unfortunate we're getting rid of this sanity check, but it seems like there's no way around it. `_modulesToTransform` was only used for this assert. Can you please delete it?